diff --git a/builder/overlay-rootfs/etc/s6/init/init-stage3 b/builder/overlay-rootfs/etc/s6/init/init-stage3
index cf41a4a..10444ba 100644
--- a/builder/overlay-rootfs/etc/s6/init/init-stage3
+++ b/builder/overlay-rootfs/etc/s6/init/init-stage3
@@ -45,14 +45,9 @@ foreground
 # Wait service down
 
 foreground { s6-echo "[s6-finish] waiting for services." }
-foreground
-{
-  backtick -D 3000 -n S6_SERVICES_GRACETIME { printcontenv S6_SERVICES_GRACETIME }
-  importas -u S6_SERVICES_GRACETIME S6_SERVICES_GRACETIME
-  elglob SERVICES /var/run/s6/services/*
-  foreground { s6-svwait -D -t ${S6_SERVICES_GRACETIME} ${SERVICES} }
-  s6-sleep -m 200
-}
+backtick -D 3000 -n S6_SERVICES_GRACETIME { printcontenv S6_SERVICES_GRACETIME }
+importas -u S6_SERVICES_GRACETIME S6_SERVICES_GRACETIME
+wait -t ${S6_SERVICES_GRACETIME} { }
 
 # Sync before TERM'n
 foreground
